src/engine/source/builder/builders/windowsHelper.cpp

```diff
@@ -0,0 +1,224 @@
+#include "windowsHelper.hpp"
+
+#include <regex>
+
+#include <baseHelper.hpp>
+
+using namespace helper::base;
+
+namespace
+{
+const std::string ACC_SID_DESC_KEY = "accountSIDDescription";
+const std::string DOM_SPC_SID_KEY = "domainSpecificSID";
+
+/**
+ * @brief Parse the list of SID, the list must be in the format:
+ *
+ * '%{sid1} %{sid2} %{sid3} ... ' // TODO: Check the format
+ *
+ * Ths function will return a vector with the sids in the same order as the input
+ * or return an empty vector if the input is not valid
+ * @param listSrt String with the list of sids
+ * @return std::vector<std::string>
+ */
+std::vector<std::string> parserListSID(const std::string& listSrt)
+{
+
+    std::vector<std::string> result = base::utils::string::split(listSrt, ' '); // TODO Check format
+
+    for (auto& sid : result)
+    {
+        if (sid.size() > 3)
+        {
+            // Remove header '%{' and tail '}'
+            sid = sid.substr(2, sid.size() - 3);
+        }
+    }
+
+    return result;
+}
+
+} // namespace
+
+namespace builder::internals::builders
+{
+HelperBuilder getWindowsSidListDescHelperBuilder(std::shared_ptr<kvdbManager::IKVDBManager> kvdb,
+                                                 const std::string& kvdbScopeName,
+                                                 std::shared_ptr<schemf::ISchema> schema)
+{
+    // TODO: add shared_ptr validation when tests are updated
+
+    return [=](const std::string& targetField,
+               const std::string& rawName,
+               const std::vector<std::string>& rawParameters,
+               std::shared_ptr<defs::IDefinitions> definitions) -> base::Expression
+    {
+        // Process parameters
+        auto parameters = processParameters(rawName, rawParameters, definitions);
+
+        // Format name
+        const auto name = formatHelperName(targetField, rawName, parameters);
+
+        // Check parameters
+        checkParametersSize(rawName, parameters, 2);
```

Comment:
magic number

Reply:
How do you propose to eliminate this magic number? This number is defined here because it is the definition of the size of the arguments